lib/usagereporter/usagereporter.go

```diff
@@ -353,16 +372,19 @@ func NewUsageReporter[T any](options *Options[T]) *UsageReporter[T] {
 			trace.Component,
 			teleport.Component(teleport.ComponentUsageReporting),
 		),
-		events:          make(chan []*SubmittedEvent[T], 1),
-		submissionQueue: make(chan []*SubmittedEvent[T], 1),
-		submit:          options.Submit,
-		clock:           options.Clock,
-		submitClock:     options.SubmitClock,
-		minBatchSize:    options.MinBatchSize,
-		maxBatchSize:    options.MaxBatchSize,
-		maxBatchAge:     options.MaxBatchAge,
-		maxBufferSize:   options.MaxBufferSize,
-		submitDelay:     options.SubmitDelay,
-		retryAttempts:   options.RetryAttempts,
+		events:                make(chan []*SubmittedEvent[T], 1),
+		submissionQueue:       make(chan []*SubmittedEvent[T], 1),
+		submit:                options.Submit,
+		clock:                 options.Clock,
+		submitClock:           options.SubmitClock,
+		minBatchSize:          options.MinBatchSize,
+		maxBatchSize:          options.MaxBatchSize,
+		maxBatchAge:           options.MaxBatchAge,
+		maxBufferSize:         options.MaxBufferSize,
+		submitDelay:           options.SubmitDelay,
+		retryAttempts:         options.RetryAttempts,
+		gracefulStopScheduled: make(chan bool),
+		gracefulStopEnqueued:  make(chan bool),
```

Comment:
AFAIK if channels are used strictly to coordinate some work and you don't really care about the values that are passed to them, an empty struct is preferred over a bool.

https://golangbyexample.com/empty-struct-in-go/